src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,114 @@
+        File localStorageFile = new File(filePath);
+        try {
+            boolean filecreated = localStorageFile.createNewFile();
+        } catch (IOException e) {
+            throw new DukeException("No local storage found");
+        } finally {
+            this.localStorageFile = localStorageFile;
+        }
+    }
+
+    /**
+     * Loads the list of tasks from local storage
+     *
+     * @return list of tasks
+     * @throws DukeException If there is no local file to be found
+     */
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> toDoList= new ArrayList<>();
+        Scanner scanner; // create a Scanner using the File as the source
+        try {
+            scanner = new Scanner(localStorageFile);
+        } catch (FileNotFoundException e) {
+            throw new DukeException("You don't have a local file, so I will create a new one just for you!");
+        }
+        while (scanner.hasNext()) {
+            String lineFromLocalStorage = scanner.nextLine();
+            if (lineFromLocalStorage.contains("[") && lineFromLocalStorage.contains("]") ) {
+                char taskType = lineFromLocalStorage.charAt(1);
+                char completed = lineFromLocalStorage.charAt(4);
+                String restOfTheTask = lineFromLocalStorage.substring(7);
+                if (taskType == 'T') {
+                    Task newTask = new ToDo(restOfTheTask);
+                    if (completed == '✅') {
+                        newTask.markAsCompleted();
+                    }
+                    toDoList.add(newTask);
+                } else if (taskType == 'E') {
+                    String[] parsedEventInput = restOfTheTask.split("\\(at: ", 2);
+                    String eventTime = parsedEventInput[1].split("\\)", 2)[0];
+                    LocalDate localDate = LocalDate.parse(eventTime, Duke.getFormatter());
+                    Task newTask = new Event(parsedEventInput[0], localDate);
+                    if (completed == '✅') {
+                        newTask.markAsCompleted();
+                    }
+                    toDoList.add(newTask);
+                } else if (taskType == 'D') {
+                    String[] parsedDeadlineInput = restOfTheTask.split("\\(by: ", 2);
+                    String deadlineTime = parsedDeadlineInput[1].split("\\)", 2)[0];
+                    LocalDate localDate = LocalDate.parse(deadlineTime, Duke.getFormatter());
+                    Task newTask = new Event(parsedDeadlineInput[0], localDate);
+                    if (completed == '✅') {
+                        newTask.markAsCompleted();
+                    }
+                    toDoList.add(newTask);
+                }
+            }
+        }
+        scanner.close();
+        return toDoList;
+    }
+
+    /**
+     * Updates the local storage everytime a change is made
+     *
+     * @param toDoList the current list of tasks
+     */
+    public void updateLs(ArrayList<Task> toDoList) {
```

Comment:
I think that this name should be `updateList` instead of `updateLs`? We were told not to use 'texting-style' spelling [here](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#use-standard-words).